### PR TITLE
docs: audit and revamp of main pages

### LINKED ISSUE #4110: Run a general language/flow audit across all docs
## Documentation

Now that that typescript-eslint.io is up, we'll want to do a general proofreading wave on the website. Much of the copy there is quite old and the context around it has aged a good bit (e.g. TSLint).

Note: I'm assigning this to myself because I'm motivated to do it, but when I send a PR it will be a _proposal_ for review.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

-   [x] Addresses an existing issue: fixes #4110
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/master/CONTRIBUTING.md) were taken

## Overview

Does a first pass of auditing the docs for flow & information duplication. Applies the following edits to most or all docs files:

* Standardizes headings within the file for capitalization/casing
* Puts sentences on their own line ([reasoning](https://github.com/DavidAnson/markdownlint/issues/81))
* Removes duplicate or unnecessary sentences

Explanations for major changes commented inline.

Does not touch the Rules section or the TODOs in plugins docs at all.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,196 +9,16 @@
...
-If you have the time and the inclination, you can even take it a step further and submit a PR to improve the project.
-
-There are also financial ways to contribute, please see [Financial Contributors](#Financial-Contributors) for more details.
-
-All positive contributions are welcome here!
+<p align="center">
+  <big>
+  See <strong><a href="https://typescript-eslint.io">typescript-eslint</a></strong> for documentation.
+  </big>
+</p>
```
Comment (JoshuaKGoldberg) [inline README.md:16]: Proposal: now that the docs page covers basic use cases, it'd be better to specify just the GitHub-area info like contributing and versioning.

File: packages/website/src/pages/index.tsx [hunk 2]
```diff
@@ -78,53 +68,53 @@ const features: FeatureItem[] = [
...
-          <img className={styles.featureImage} src={imgUrl} alt={title} />
-        </div>
-      )}
       <h2 className="text--center">{title}</h2>
       {description}
+      <div className={styles.buttons}>
+        <Link className="button button--primary" to={useBaseUrl('docs/')}>
+          Get Started
+        </Link>
+      </div>
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:117]: Learning from working in product land: you want a CTA (Call To Action) visible on the screen at all times. Even if it's as simple as repeating the same basic button below each section, it helps make sure users both always know what to do and are visually encouraged to do it.

File: docs/getting-started/linting/ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+```
+
+ESLint uses an AST format known as **[`estree`]**.
+
+ESTree is more broadly used than just for ESLint -- it is the de facto community standard.
+ESLint' built-in parser that outputs an `estree`-shaped AST is also a separate package, called **[`espree`]**.
+
+:::note
+You can play more with various ASTs such as ESTree on [astexplorer.net] and read more details on their [Wikipedia article](https://en.wikipedia.org/wiki/Abstract_syntax_tree).
```
Comment (JoshuaKGoldberg) [inline docs/getting-started/linting/ARCHITECTURE.md:46]: Build failure: https://github.com/typescript-eslint/typescript-eslint/runs/4148557257?check_suite_focus=true

```
/home/runner/work/typescript-eslint/typescript-eslint/docs/getting-started/linting/ARCHITECTURE.md:46:56 - Unknown word (astexplorer)
```

Will need to get these into the cspell dictionary soon. Will do tomorrow.

File: docs/getting-started/linting/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -164,25 +161,34 @@ If you're experiencing times much slower than that, then there are a few common
...
+This plugin surfaces prettier formatting problems at lint time, helping to ensure your code is always formatted.
+However this comes at a quite a large cost - in order to figure out if there is a difference, it has to do a prettier format on every file being linted.
+This means that each file will be parsed twice - once by ESLint, and once by Prettier.
+This can add up for large codebases.
 
-Instead of using this plugin, we recommend using prettier's `--list-different` flag to detect if a file has not been correctly formatted. For example, our CI is setup to run the following command automatically, which blocks diffs that have not been formatted:
+Instead of using this plugin, we recommend using prettier's `--list-different` flag to detect if a file has not been correctly formatted.
+For example, our CI is setup to run the following command automatically, which blocks diffs that have not been formatted:
```
Comment (bradzacher) [inline docs/getting-started/linting/TROUBLESHOOTING.md:181]: oops.. `diffs` is a facebookism.

```suggestion
For example, our CI is setup to run the following command automatically, which blocks PRs that have not been formatted:
```

File: packages/website/sidebars/sidebar.base.js [hunk 5]
```diff
@@ -11,6 +11,7 @@ module.exports = {
           'getting-started/linting/type-linting',
           'getting-started/linting/monorepo',
           'getting-started/linting/troubleshooting',
+          'getting-started/linting/architecture',
```
Comment (bradzacher) [inline packages/website/sidebars/sidebar.base.js:14]: Question: I wonder if this belongs in a larger "about us" set of pages?

It's probs fine living here until we build out a few about pages.

File: packages/website/src/pages/index.tsx (hunk 2 above)
Comment (bradzacher) [reply packages/website/src/pages/index.tsx:117]: bro, do you even get started?

image

**Review by bradzacher — COMMENTED** [review]
Overall looks fantastic!



One thing I noticed you doing here is adding a lot of newlines.
Just checking: were you trying to add line breaks into the output, or were you just breaking the file up a bit?

If it's the former - docasaurus doesn't use "github flavoured markdown" - so single newlines aren't respected. You need to terminate a line with two spaces in order for it to respect the newline.

If it's the latter - carry on good sir.

File: packages/website/sidebars/sidebar.base.js (hunk 5 above)
Comment (JoshuaKGoldberg) [reply packages/website/sidebars/sidebar.base.js:14]: I dig it - maybe architecture, contributors, sponsors, financial support? 

Added a card to the [board](https://github.com/orgs/typescript-eslint/projects/1/views/1) named _"Flesh out "About Us" section in the site_.

**Comment by JoshuaKGoldberg** [comment]
> or were you just breaking the file up a bit

@bradzacher yeah, I have a strong, maybe overly opinionated preference for having a single sentence per line. IMO it makes for better diffs. It's something I'll probably bring up enforcing eventually 😄

**Review by bradzacher — APPROVED** [review]
lets goooo

**Pull request merged**